learning_journal/security.py

```diff
@@ -0,0 +1,46 @@
+import os
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from pyramid.security import Allow
+from pyramid.security import Everyone, Authenticated
+from passlib.apps import custom_app_context as pwd_context
+from pyramid.session import SignedCookieSessionFactory
+
+
+class TheRoot(object):
+    def __init__(self, request):
+        self.request = request
+
+    __acl__ = [
+        (Allow, Authenticated, 'amend'),
+        (Allow, Everyone, 'view'),
+    ]
+
+
+def check_credentials(username, password):
+    """Return True if correct username and password, else False."""
+    stored_username = os.environ.get('AUTH_USERNAME', '')
+    stored_password = os.environ.get('AUTH_PASSWORD', '')
+    is_authenticated = False
+    if stored_username and stored_password:
+        if username == stored_username:
+            try:
+                is_authenticated = pwd_context.verify(password, stored_password)
+            except ValueError:
+                # ValueError is raised if the stored password is not hashed or if the salt is improper
```

Comment:
OK, I got rid of the try-except, thinking if it errors, no reason to flag it. Is that right, or should there be a an error raised?